geode-apis-compatible-with-redis/src/distributedTest/java/org/apache/geode/redis/SSLDUnitTest.java

```diff
@@ -0,0 +1,301 @@
+  }
+
+  @Test
+  public void givenServerCertificateIsRotated_clientCanStillConnect() throws Exception {
+    String newServerName = "updated-server";
+
+    try (Jedis jedis = createClient(true, false)) {
+      assertThat(jedis.ping()).isEqualTo("PONG");
+    }
+
+    // create a new certificate for the server
+    CertificateMaterial serverCertificate = new CertificateBuilder()
+        .commonName(newServerName)
+        .issuedBy(ca)
+        .sanDnsName(newServerName)
+        .generate();
+
+    CertStores serverStore = CertStores.serverStore();
+    serverStore.withCertificate("server", serverCertificate);
+    serverStore.trust("ca", ca);
+
+    // Wait for one second since file timestamp granularity may only be seconds depending on the
+    // platform.
+    GeodeAwaitility.await().during(Duration.ofSeconds(1)).until(() -> true);
+    serverStore.createKeyStore(serverKeyStoreFilename, commonPassword);
+
+    // Try long enough for the file change to be detected
+    GeodeAwaitility.await().atMost(Duration.ofSeconds(PollingFileWatcher.PERIOD_SECONDS * 2))
+        .untilAsserted(() -> {
+          try (Jedis jedis = createClient(true, false)) {
+            jedis.ping();
+            assertThat(hostnameVerifier.getSubjectAltNames()).contains(newServerName);
+          }
+        });
+  }
+
+  @Test
+  public void givenServerCAandKeyIsRotated_clientCannotConnect() throws Exception {
+    try (Jedis jedis = createClient(true, false)) {
+      assertThat(jedis.ping()).isEqualTo("PONG");
+    }
+
+    CertificateMaterial newCA = new CertificateBuilder()
+        .commonName("New Test CA")
+        .isCA()
+        .generate();
+
+    CertificateMaterial serverCertificate = new CertificateBuilder()
+        .commonName("server")
+        .issuedBy(newCA)
+        .generate();
+
+    CertStores serverStore = CertStores.serverStore();
+    serverStore.withCertificate("server", serverCertificate);
+    serverStore.trust("ca", newCA);
+
+    // Wait for one second since file timestamp granularity may only be seconds depending on the
+    // platform.
+    GeodeAwaitility.await().during(Duration.ofSeconds(1)).until(() -> true);
+    serverStore.createKeyStore(serverKeyStoreFilename, commonPassword);
+    serverStore.createTrustStore(serverTrustStoreFilename, commonPassword);
+
+    IgnoredException.addIgnoredException(SSLHandshakeException.class);
+    IgnoredException.addIgnoredException("SunCertPathBuilderException");
+
+    // Try long enough for the file change to be detected
+    GeodeAwaitility.await().atMost(Duration.ofSeconds(PollingFileWatcher.PERIOD_SECONDS * 2))
+        .untilAsserted(() -> {
+          assertThatThrownBy(() -> createClient(true, false))
+              .isInstanceOf(JedisConnectionException.class);
+        });
+
+    IgnoredException.removeAllExpectedExceptions();
+  }
+
+  private Jedis createClient(boolean mutualAuthentication, boolean isSelfSigned) throws Exception {
```

Comment:
I didn't see `isSelfSigned` being set to true anywhere in the test.

Reply:
There are a couple of tests that do so.